File: src/main/java/rustleund/fightingfantasy/framework/closures/SetPoisonImmunity.java
package rustleund.fightingfantasy.framework.closures;

import org.w3c.dom.Element;

import rustleund.fightingfantasy.framework.AbstractCommand;
import rustleund.fightingfantasy.framework.GameState;

public class SetPoisonImmunity extends AbstractCommand {

	private boolean immune;

	public SetPoisonImmunity(Element element) {
		this.immune = attributeValue(element, "immune", true);
	}

	public void execute(GameState gameState) {
		gameState.getPlayerState().setPoisonImmunity(this.immune);
	}

}

File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/SetPoisonImmunity.kt
package rustleund.fightingfantasy.framework.closures.impl

import org.w3c.dom.Element
import rustleund.fightingfantasy.framework.base.GameState
import rustleund.fightingfantasy.framework.base.booleanAttribute
import rustleund.fightingfantasy.framework.closures.Closure

class SetPoisonImmunity(element: Element) : Closure {

    private val immune: Boolean = element.booleanAttribute("immune", true)

    override fun execute(gameState: GameState): Boolean {
        gameState.playerState.isPoisonImmunity = immune
        return true
    }
}